File: Source/Fayde.Server/Media/VSM/VisualTransition.cs
using Fayde.Core;
using Fayde.Media.Animation;
using Fayde.Primitives;
using Fayde.Xaml.Metadata;

namespace Fayde.Media.VSM
{
    public class VisualTransition : DependencyObject
    {
        public static readonly PropertyDescription StoryboardProperty = PropertyDescription.Register("Storyboard", typeof(Storyboard), typeof(VisualTransition), true);
        public static readonly PropertyDescription FromProperty = PropertyDescription.Register("From", typeof(string), typeof(VisualTransition));
        public static readonly PropertyDescription ToProperty = PropertyDescription.Register("To", typeof(string), typeof(VisualTransition));
        public static readonly PropertyDescription GeneratedDurationProperty = PropertyDescription.Register("GeneratedDuration", typeof(Duration), typeof(VisualTransition));
    }
}

File: Source/Fayde.Client/Media/VSM/VisualTransition.ts
/// <reference path="../../Core/DependencyObject.ts" />

module Fayde.Media.VSM {
    export class VisualTransition extends DependencyObject {
        From: string = null;
        To: string = null;
        Storyboard: Animation.Storyboard;
        GeneratedDuration: Duration;
        DynamicStoryboardCompleted: boolean = true;
        ExplicitStoryboardCompleted: boolean = true;
        GeneratedEasingFunction: Animation.EasingFunctionBase;
        get IsDefault(): boolean { return this.From == null && this.To == null; }
    }
    Fayde.RegisterType(VisualTransition, "Fayde.Media.VSM", Fayde.XMLNS);
}